app/src/main/java/org/simple/clinic/appupdate/criticalupdatedialog/CriticalAppUpdateUiRenderer.kt

```diff
@@ -15,11 +15,15 @@ class CriticalAppUpdateUiRenderer(
       ui.hideHelp()
     }
 
-    if (model.isCriticalUpdateNudgePriority && model.hasAppStaleness)
-      ui.renderCriticalAppUpdateReason(appStalenessInMonths(model.appStaleness!!))
+    renderCriticalAppUpdateReason(model)
+  }
 
-    if (model.isCriticalSecurityUpdateNudgePriority)
+  private fun renderCriticalAppUpdateReason(model: CriticalAppUpdateModel) {
+    if (model.isCriticalUpdateNudgePriority && model.hasAppStaleness) {
+      ui.renderCriticalAppUpdateReason(appStalenessInMonths(model.appStaleness!!))
+    } else if (model.isCriticalSecurityUpdateNudgePriority) {
       ui.renderCriticalSecurityAppUpdateReason()
+    }
```

Comment:
Should we throw an error in `else` condition here? If there is an else condition we currently don't render any text and show empty text space.

Reply:
Honestly, that would be over optimising for a situation that's never gonna happen.

Reply:
This dialog will only be shown when there is an app update nudge priority and only when it is critical or critical security

Reply:
Hmm, I don't think it's over-optimizing in this scenario, since it fails silently. For example, whenever we are using a type in a conditional we either have a "default" state in the else condition or crash it. But yeah, this might not be updated as often. I guess we can leave it as it is.